src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.Win32.Unix.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+namespace System.IO.IsolatedStorage
+{
+    internal static partial class Helper
+    {
+        internal static string GetDataDirectory(IsolatedStorageScope scope)
+        {
+            // This is the relevant special folder for the given scope plus "IsolatedStorage".
+            // It is meant to replicate the behavior of the VM ComIsolatedStorage::GetRootDir().
+
+            // (note that Silverlight used "CoreIsolatedStorage" for a directory name and did not support machine scope)
+
+            string dataDirectory = null;
+
+            if (IsMachine(scope))
+            {
+                // SpecialFolder.CommonApplicationData -> C:\ProgramData
+                dataDirectory = Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData);
+            }
+            else
+            {
+                if (!IsRoaming(scope))
```

Comment:
Nit: rather than a not condition, I'd personally have an easier time reading this if it was:
```C#
if (IsMachine(scope))
{
    ...
}
else if (IsRoaming(scope))
{
    ...
}
else
{
    ....
}
```